core/src/main/java/org/elasticsearch/plugins/SearchPlugin.java

```diff
@@ -99,6 +109,38 @@ public ScoreFunctionSpec(String name, Reader<T> reader, ScoreFunctionParser<T> p
     }
 
     /**
+     * Specification of custom {@link Query}.
+     */
+    public class QuerySpec<T extends QueryBuilder> extends SearchPluginSpec<T, QueryParser<T>> {
```

Comment:
I like calling it SearchExtensionSpec and will do the rename. I made the
subclass because the builder and parser have types that line up. So we can
enforce that you register a sensible builder and parser combination.

On Jul 20, 2016 8:47 AM, "Simon Willnauer" <notifications@github.com> wrote:

> In core/src/main/java/org/elasticsearch/plugins/SearchPlugin.java
> <https://github.com/elastic/elasticsearch/pull/19376#discussion_r71517232>
> :
>
> > @@ -99,6 +109,38 @@ public ScoreFunctionSpec(String name, Reader<T> reader, ScoreFunctionParser<T> p
> >      }
> >
> >      /**
> > +     * Specification of custom {@link Query}.
> > +     */
> > +    public class QuerySpec<T extends QueryBuilder> extends SearchPluginSpec<T, QueryParser<T>> {
>
> can't we simply return SearchPluginSpec instead of having yet another
> class for this? Maybe call it SearchExtensionSpec instead
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/19376/files/6e8f16d2a7a53f64264dc6c885c84763c2d58d0c#r71517232>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANLokAcyvJWEYDjF90R2-xpte0G4neUks5qXhjHgaJpZM4JJ9Yo>
> .
>